pkg/controller/core/reconciler.go

```diff
@@ -0,0 +1,457 @@
+			claim.SetConditions(v1alpha1.ReconcileError(err))
+			return reconcile.Result{RequeueAfter: AShortWait}, errors.Wrap(r.client.Status().Update(ctx, claim), "cannot update resource claim status")
+		}
+	}
+
+	if meta.WasDeleted(claim) {
+		if err := r.resource.Delete(ctx, claim, resource); err != nil {
+			// If we didn't hit this error last time we'll be requeued
+			// implicitly due to the status update. Otherwise we want to retry
+			// after a brief wait, in case this was a transient error.
+			claim.SetConditions(v1alpha1.Deleting(), v1alpha1.ReconcileError(err))
+			return reconcile.Result{RequeueAfter: AShortWait}, errors.Wrap(r.client.Status().Update(ctx, claim), "cannot update resource claim status")
+		}
+
+		// We've successfully processed the delete, so there's no further
+		// reconciliation to do. There's a good chance our claim no longer
+		// exists, but we try update its status just in case it sticks around,
+		// for example due to additional finalizers.
+		claim.SetConditions(v1alpha1.Deleting(), v1alpha1.ReconcileSuccess())
+		return reconcile.Result{Requeue: false}, errors.Wrap(IgnoreNotFound(r.client.Status().Update(ctx, claim)), "cannot update resource claim status")
+	}
+
+	if !meta.WasCreated(resource) {
+		class := &v1alpha1.ResourceClass{}
+		if err := r.client.Get(ctx, meta.NamespacedNameOf(claim.GetClassReference()), class); err != nil {
+			// If we didn't hit this error last time we'll be requeued
+			// implicitly due to the status update. Otherwise we want to retry
+			// after a brief wait, in case this was a transient error or the
+			// class is (re)created.
+			claim.SetConditions(v1alpha1.Creating(), v1alpha1.ReconcileError(err))
+			return reconcile.Result{RequeueAfter: AShortWait}, errors.Wrap(r.client.Status().Update(ctx, claim), "cannot update resource claim status")
+		}
+
+		if err := r.resource.Configure(ctx, claim, class, resource); err != nil {
+			// If we didn't hit this error last time we'll be requeued
+			// implicitly due to the status update. Otherwise we want to retry
+			// after a brief wait, in case this was a transient error or some
+			// issue with the resource class was resolved.
+			claim.SetConditions(v1alpha1.Creating(), v1alpha1.ReconcileError(err))
+			return reconcile.Result{RequeueAfter: AShortWait}, errors.Wrap(r.client.Status().Update(ctx, claim), "cannot update resource claim status")
+		}
+
+		if err := r.resource.Create(ctx, claim, class, resource); err != nil {
+			// If we didn't hit this error last time we'll be requeued
+			// implicitly due to the status update. Otherwise we want to retry
+			// after a brief wait, in case this was a transient error.
+			claim.SetConditions(v1alpha1.Creating(), v1alpha1.ReconcileError(err))
+			return reconcile.Result{RequeueAfter: AShortWait}, errors.Wrap(r.client.Status().Update(ctx, claim), "cannot update resource claim status")
+		}
+	}
+
+	if resource.GetBindingPhase() == v1alpha1.BindingPhaseUnbindable {
```

Comment:
i'm not sure i understand the meaning of this condition, what does it mean for a resource to be unbindable? that sounds like a bad thing, but we seem to be calling this a success and no longer requeueing.

Reply:
The intent behind this new 'unbindable' phase was to communicate whether the concrete resource is ready to be bound, as an alternative to inspecting either its conditioned status or other status fields that vary from resource to resource.

My thought was that a resource would be either:
* Unbindable, i.e. still creating, or perhaps broken in some way.
* Unbound, i.e. ready for binding
* Bound, i.e. bound to a claim.

I considered Pending as well rather than Unbindable. Do you think I could make this clearer somehow? It seems like at least I should add Godoc comments to each phase.